docs/artifacts/includes/nuget/nuget-publish-endpoint.md

```diff
@@ -18,14 +18,78 @@ ms.date: 02/18/2022
 
     :::image type="content" source="../../media/nuget-connect-feed.png" alt-text="NuGet.exe feed connection":::
 
-1. If this is the first time using Azure Artifacts with Nuget.exe, select **Get the tools** button and follow the instructions to install the prerequisites.
+1. When using a local development environment, it is recommended to have the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) installed which provides authentication when pushing or downloading packages (see examples section).
 
     1. Download the [latest NuGet version](https://www.nuget.org/downloads).
-    1. Download and install the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider#azure-artifacts-credential-provider).
 
-1. Follow the instructions in the **Project setup** to add a config file. 
+    1. Follow the instructions in the **Project setup** to add a config file. 
+
+        :::image type="content" source="../../media/project-setup.png" alt-text="Project setup":::
+
+1. For Azure hosted builds that interact with Azure Artifacts feeds, you must use the [NuGet Authenticate](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget-authenticate) task for the authentication(see examples section).
+
+1. For Github actions builds that interact with Azure Artifacts feeds, you may use [Github setup dotnet action](https://github.com/actions/setup-dotnet) for the authentication(see examples section).
+
+1. For other hosted builds that interact with Azure Artifacts feeds, you may use [Azure PAT]([url](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows)) saved in `nuget.config` file for the authentication(see examples section), but please exercise caution because it may leak your PAT as part of build environement log collection or source control checkout, artifact upload accidentally.
```

Comment:
I added `other hosted builds` section because there're other hosting such as Teamcity, Jenkins etc which may don't support [NuGet Authenticate](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget-authenticate) task, but still want to publish nupkg package to Azure Artifacts.